python/dlisio/__init__.py

```diff
@@ -357,27 +314,18 @@ def unknowns(self):
 
         Returns
         -------
-        objects : defaultdict(list)
+        objects : defaultdict(dict)
             A defaultdict index by object-type
 
         """
-        recs = [rec for rec, t in zip(self.attic, self.record_types)
-            if  t not in self.types
-            and not rec.encrypted
-            and t not in self.indexedobjects
-        ]
-
-        self.load(recs, reload=False)
-
-        unknowns = defaultdict(list)
-
-        for key, value in self.indexedobjects.items():
-            if key in self.types: continue
-            unknowns[key] = value
+        unknowns = defaultdict(dict)
+        for t in set(self.metadata.types):
```

Comment:
Would be cool to have tests of kind
types = [Channel, My_set, Frame, My_set] 
to see that all objects from My_set are present